go/datas/commit_test.go

```diff
@@ -82,3 +84,57 @@ func TestCommitWithoutMetaField(t *testing.T) {
 	})
 	assert.False(IsCommitType(noMetaCommit.Type()))
 }
+
+func TestCommitDescendsFrom(t *testing.T) {
+	assert := assert.New(t)
```

Comment:
high-level issue: I think this test would be clearer if it first setup a DAG of commits which collected local variables named `a`, `b`, `c`, etc... (each of these would be a `Ref<Commit>`) and some ascii-art in a comment which depicts the graph, something like: 

https://github.com/attic-labs/noms/blob/master/go/datas/pull_test.go#L168

Then at the bottom a bunch of assertions for the set of cases you need to test, e.g.

`assert.True(descendsFrom(a, b))`
`assert.False(descendsFrom(c, e))`

etc...